.github/workflows/codeql-analysis.yml

```diff
@@ -32,6 +36,10 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v2
 
+    - uses: whelk-io/maven-settings-xml-action@v15
```

Comment:
Did you review this action? Custom actions needs to be thoroughly reviewed to make sure they are not doing something they shouldn't

Reply:
Good point. I took the same approach as in #8007.
@vmuzikar Have you reviewed this action in detail?

Reply:
I haven't done a line-by-line review, but the action seems pretty simple and straightforward. First, it [prepares in-memory maven xml](https://github.com/whelk-io/maven-settings-xml-action/blob/v15/src/settings.js) based on a [predefined template](https://github.com/whelk-io/maven-settings-xml-action/blob/v15/template/settings.xml), and then it [writes it to the .m2 dir](https://github.com/whelk-io/maven-settings-xml-action/blob/v15/src/index.js). Doesn't seem dangerous.

Reply:
What the action says it does and what it does can be two very different things, and the action could also be compromised down the road.

Recommendations here are to use SHA and not version as a version can be modified. Also, to carefully review all actions that are not published by an already trusted source like GitHub. We haven't done this properly in the past, but should start doing it. Usually an action is only a few lines of code, so doesn't take much time to scan through.

Reply:
I took a different approach without a need for external actions. Could you please re-review?